mm2src/lp_bot.rs

```diff
@@ -0,0 +1,98 @@
+//
+//  lp_bot.rs
+//  marketmaker
+//
+
+use common::mm_ctx::{from_ctx, MmArc};
+use derive_more::Display;
+use futures::lock::Mutex as AsyncMutex;
+use std::{collections::HashMap, sync::atomic::AtomicBool, sync::Arc};
+
+#[cfg(test)] use mocktopus::macros::*;
+
+#[path = "lp_bot/simple_market_maker.rs"]
+mod simple_market_maker_bot;
+pub use simple_market_maker_bot::{start_simple_market_maker_bot, stop_simple_market_maker_bot};
+
+#[cfg(all(test, not(target_arch = "wasm32")))]
+#[path = "lp_bot_tests.rs"]
+mod lp_bot_tests;
+
+#[derive(Default)]
+struct TradingBotStates {
+    /// Used to determine if the bot is running
+    pub is_running: AtomicBool,
+
+    /// Used to determine if the bot is shutting down
+    pub is_stopping: AtomicBool,
+}
+
+pub type SimpleMakerBotRegistry = HashMap<String, SimpleCoinMarketMakerCfg>;
+
+#[derive(Debug, Serialize, Deserialize, Display, Clone)]
+#[display(fmt = "{} {} {} {}", base, rel, min_volume, spread)]
+pub struct SimpleCoinMarketMakerCfg {
+    base: String,
+    rel: String,
+    min_volume: String,
+    spread: String,
+    base_confs: i64,
+    base_nota: bool,
+    rel_confs: i64,
+    rel_nota: bool,
+    enable: bool,
+    price_elapsed_validity: Option<f64>,
+    check_last_bidirectional_trade_thresh_hold: Option<bool>,
+    max: Option<bool>,
+    balance_percent: Option<String>,
```

Comment:
String type doesn't look suitable for the field called `balance_percent`. I assume it should have a numeric type.

Reply:
Should be solved in https://github.com/KomodoPlatform/atomicDEX-API/pull/1066/commits/8ef6883858cee013cfc79e08dbaa3461c494c261